fe/fe-core/src/main/java/com/starrocks/memory/MemoryUsageTracker.java

```diff
@@ -0,0 +1,131 @@
+// limitations under the License.
+
+package com.starrocks.memory;
+
+import com.google.common.collect.Maps;
+import com.starrocks.common.Config;
+import com.starrocks.common.util.FrontendDaemon;
+import com.starrocks.common.util.ProfileManager;
+import com.starrocks.monitor.unit.ByteSizeValue;
+import com.starrocks.qe.QeProcessor;
+import com.starrocks.qe.QeProcessorImpl;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.optimizer.statistics.CacheDictManager;
+import com.starrocks.sql.optimizer.statistics.IDictManager;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MemoryUsageTracker extends FrontendDaemon {
+
+    private static final Logger LOG = LogManager.getLogger(MemoryUsageTracker.class);
+
+    // Used to save references to memory objects that need to be counted.
+    // If the object needs to be counted, it first needs to be added to this collection.
+    private static final Map<String, Map<String, MemoryTrackable>> REFERENCE = Maps.newConcurrentMap();
+
+    private static final Map<String, MemoryStat> MEMORY_USAGE = Maps.newConcurrentMap();
+
+    private boolean initialize;
+    public MemoryUsageTracker() {
+        super("MemoryUsageTracker", Config.memory_tracker_interval_seconds * 1000L);
+    }
+
+    private void initMemoryTracker() {
+        GlobalStateMgr currentState = GlobalStateMgr.getCurrentState();
+
+        addReference("Load", currentState.getLoadMgr());
+        addReference("Load", currentState.getRoutineLoadMgr());
+        addReference("Load", currentState.getStreamLoadMgr());
+
+        addReference("Export", currentState.getExportMgr());
+        addReference("Delete", currentState.getDeleteMgr());
+        addReference("Transaction", currentState.getGlobalTransactionMgr());
+        addReference("Backup", currentState.getBackupHandler());
+        addReference("Task", currentState.getTaskManager());
+        addReference("Task", currentState.getTaskManager().getTaskRunManager());
+        addReference("Tablet", currentState.getTabletInvertedIndex());
+        addReference("Profile", ProfileManager.getInstance());
+        addReference("Partition", new PartitionMemoryTracker());
+        addReference("ExternalCache", new PartitionMemoryTracker());
+
+        QeProcessor qeProcessor = QeProcessorImpl.INSTANCE;
+        if (qeProcessor instanceof QeProcessorImpl) {
+            addReference("Coordinator", (QeProcessorImpl) qeProcessor);
+        }
+
+        IDictManager dictManager = IDictManager.getInstance();
+        if (dictManager instanceof CacheDictManager) {
+            addReference("Dict", (CacheDictManager) dictManager);
+        }
+
+
+        LOG.info("Memory usage tracker init success");
+
+        initialize = true;
+    }
+
+    public static void registerMemoryReference(String moduleName, MemoryTrackable object) {
```

Comment:
You can use MemoryUsageTracker.registerMemoryReference to add a reference anywhere, but it may be better to make MemoryUsageTracker a singleton, and then you can get this singleton object anywhere, so registerMemoryReference does not need to be static.